### PR TITLE
Fix ip output

### PR DESCRIPTION
**Pull request by baristaTam**
### Changed

- IP logic in Get-QualysAssetGroups sometimes resulted in a string being returned instead of a string array for the IPs property

### CODE REVIEW

**Review by mabaumgartner — APPROVED** [review]

**Review by mabaumgartner — DISMISSED** [review]

File: src/UofIQualys/functions/public/Get-QualysAssetGroups.ps1 [hunk 1]
```diff
@@ -49,19 +49,24 @@ function Get-QualysAssetGroups{
             }
         }
 
-        $IPs = @()
         $Response = Invoke-QualysRestCall @RestSplat
+        #This will return IP information for every asset group
         if(!($Identity)){
             $Index = 0
-            foreach ($ID in $Response.ASSET_GROUP_LIST_OUTPUT.RESPONSE.ASSET_GROUP_LIST.ASSET_GROUP.ID ) {
+            $AssetGroupIDs = $Response.ASSET_GROUP_LIST_OUTPUT.RESPONSE.ASSET_GROUP_LIST.ASSET_GROUP.ID
```
Comment (edthedev) [inline src/UofIQualys/functions/public/Get-QualysAssetGroups.ps1:56]:
```suggestion
$AssetGroups = $Response.ASSET_GROUP_LIST_OUTPUT.RESPONSE.ASSET_GROUP_LIST.ASSET_GROUP
```

File: src/UofIQualys/functions/public/Get-QualysAssetGroups.ps1 [hunk 2]
```diff
@@ -49,19 +49,24 @@ function Get-QualysAssetGroups{
...
+            foreach ($ID in $AssetGroupIDs) {
```
Comment (edthedev) [inline src/UofIQualys/functions/public/Get-QualysAssetGroups.ps1:57]:
```suggestion
foreach ($ID in $AssetGroups.ID) {
```

File: src/UofIQualys/functions/public/Get-QualysAssetGroups.ps1 [hunk 3]
```diff
@@ -49,19 +49,24 @@ function Get-QualysAssetGroups{
...
+                $IPs = @()
+                #If there are ranges get the ranges and add them to the IP array
                 If($Response.ASSET_GROUP_LIST_OUTPUT.RESPONSE.ASSET_GROUP_LIST.ASSET_GROUP.IP_SET[$Index].IP_RANGE){
```
Comment (edthedev) [inline src/UofIQualys/functions/public/Get-QualysAssetGroups.ps1:60]:
```suggestion
If($AssetGroups.IP_SET[$Index].IP_RANGE){
```

and then we can make this change for each of the `if` lines below as well.

File: src/UofIQualys/functions/public/Get-QualysAssetGroups.ps1 [hunk 4]
```diff
@@ -49,19 +49,24 @@ function Get-QualysAssetGroups{
...
-                    $IPs = $Response.ASSET_GROUP_LIST_OUTPUT.RESPONSE.ASSET_GROUP_LIST.ASSET_GROUP.IP_SET[$Index].IP_RANGE
+                    $IPs += $Response.ASSET_GROUP_LIST_OUTPUT.RESPONSE.ASSET_GROUP_LIST.ASSET_GROUP.IP_SET[$Index].IP_RANGE
+                    #If there are IPs and ranges now add the IPs to the array with the ranges
                     If($Response.ASSET_GROUP_LIST_OUTPUT.RESPONSE.ASSET_GROUP_LIST.ASSET_GROUP.IP_SET[$Index].IP){
```
Comment (edthedev) [inline src/UofIQualys/functions/public/Get-QualysAssetGroups.ps1:63]:
```suggestion
If($AssetGroups.IP_SET[$Index].IP){
```

**Review by edthedev — CHANGES_REQUESTED** [review]
I would like to see the change I suggested below made to increase readability.

**Review by edthedev — APPROVED** [review]

**Review by mabaumgartner — APPROVED** [review]

**Pull request merged**